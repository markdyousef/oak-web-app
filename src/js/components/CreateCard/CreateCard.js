// @flow
import React, { Component } from 'react';
import styled from 'styled-components';
import { colors } from '../../styles';
import LabelsActionBoxContainer from '../../containers/LabelsActionBoxContainer';
import Dropdown from '../shared/Dropdown';

const Container = styled.div`
    width: 200px;
    ${''/* height: 100px; */}
    & h3 {
        font-size: 18px;
    }
`;

const Select = styled.select`
    margin: 16px 0;
    width: 100%;
    width: 100%;
    border: 1px solid #e5e5e5;
    background: #fff;
    appearance: none;
    padding: 12px;
    font-size: 14px;
    font-weight: bold;
    border-radius: 3px;
    outline: none;
    background-image: url(../../../img/dropdown-fat.svg);
    background-size: 16px;
    background-position: 50% 90%;
    cursor: pointer;
`;

const Labels = styled.div`
    position: relative;
    & button {
        width: 100%;
    }
`;


const Footer = styled.div`
    border-top: 1px solid ${colors.lightGrey};
    width: 100%;
    height: 50px;
    position: relative;
    & button {
        position: absolute;
        right: 0;
        margin-top: 20px;
        color: ${colors.green};
        border: none;
        padding: 0;
        background-color: ${colors.white};
        font-size: 14px;
        cursor: pointer;
    }
`;

const LabelsContainer = styled.div`
    position: absolute;
    right: 0;
    z-index: 9999;
    width: 250px;
`;


type DefaultProps = {
    onShowLabels: () => void
};

type Props = {
    collectionId?: string,
    collections?: Array<Object>,
    updateCollection?: (id: string) => void,
    addCard?: Function,
    showLabels?: bool,
    onShowLabels: () => void
};

type State = {
    collection: ?string
};

export default class CreateCard extends Component<DefaultProps, Props, State> {
    static defaultProps: DefaultProps = {
        onShowLabels: () => {}
    };
    props: Props;
    state: State;
    addCard = () => {
        const { addCard, collectionId } = this.props;
        if (addCard) {
            addCard(collectionId);
        }
    }
    changeCollection = (id: string) => {
        const { updateCollection } = this.props;
        if (updateCollection) updateCollection(id);
    }
    render() {
        const { collections, addCard, collectionId, showLabels, onShowLabels } = this.props;
        return (
            <Container>
                <h3>Share post in:</h3>
                <Select
                    value={collectionId || ''}
                    onChange={event => this.changeCollection(event.target.value)}
                >
                    <option value="" selected>Select collection</option>
                    {collections && collections.map(item =>
                        <option
                            key={item.id}
                            value={item.id}
                        >
                            {item.name}
                        </option>
                    )}
                </Select>
                {collectionId &&
                    <Labels>
                        <button onClick={() => onShowLabels()}>LABELS</button>
                        {showLabels &&
                            <LabelsContainer>
                                <Dropdown onClose={() => onShowLabels(false)} arrowPos="none">
                                    <LabelsActionBoxContainer />
                                </Dropdown>
                            </LabelsContainer>
                        }
                    </Labels>
                }
                <Footer>
                    {collectionId &&
                    <button onClick={this.addCard}>
                        Share
                    </button>}
                </Footer>
            </Container>
        );
    }
}
