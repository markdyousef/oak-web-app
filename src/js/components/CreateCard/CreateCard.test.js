// @flow
import React from 'react';
import { shallow } from 'enzyme';
import CreateCard from './CreateCard';

it('renders correctly', () => {
    const props = {
    };
    const shallowComponent = shallow(
        <CreateCard {...props} />
    );
    expect(shallowComponent).toMatchSnapshot();
});
