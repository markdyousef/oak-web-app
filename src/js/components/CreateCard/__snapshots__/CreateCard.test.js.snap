exports[`test renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.div>
    <h3>
        Share post in:
    </h3>
    <styled.select
        onChange={[Function]}
        value="">
        <option
            selected={true}
            value="">
            Select collection
        </option>
    </styled.select>
    <styled.div>
        You can add labels to your post once you’ve selected a collection.
    </styled.div>
    <styled.div />
</styled.div>,
  "nodes": Array [
    <styled.div>
      <h3>
            Share post in:
      </h3>
      <styled.select
            onChange={[Function]}
            value="">
            <option
                  selected={true}
                  value="">
                  Select collection
            </option>
      </styled.select>
      <styled.div>
            You can add labels to your post once you’ve selected a collection.
      </styled.div>
      <styled.div />
</styled.div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <CreateCard
        onShowLabels={[Function]} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": CreateCard {
        "_reactInternalInstance": [Circular],
        "addCard": [Function],
        "changeCollection": [Function],
        "context": Object {},
        "props": Object {
          "onShowLabels": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.div>
          <h3>
                    Share post in:
          </h3>
          <styled.select
                    onChange={[Function]}
                    value="">
                    <option
                              selected={true}
                              value="">
                              Select collection
                    </option>
          </styled.select>
          <styled.div>
                    You can add labels to your post once you’ve selected a collection.
          </styled.div>
          <styled.div />
</styled.div>,
        "_debugID": 2,
        "_renderedOutput": <styled.div>
          <h3>
                    Share post in:
          </h3>
          <styled.select
                    onChange={[Function]}
                    value="">
                    <option
                              selected={true}
                              value="">
                              Select collection
                    </option>
          </styled.select>
          <styled.div>
                    You can add labels to your post once you’ve selected a collection.
          </styled.div>
          <styled.div />
</styled.div>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CreateCard
    onShowLabels={[Function]} />,
}
`;
